data_warehouse/services/table_export_service.py

```diff
@@ -342,6 +343,65 @@ def export_training_feedback(cls, data):
             worksheet.write(ptr_r, 4, item['feedback_user_name'], style)
             worksheet.write(ptr_r, 5, item['feedback_user_department'], style)
             worksheet.write(ptr_r, 6, item['feedback_user_email'], style)
+        # 写入数据
+        _, file_path = tempfile.mkstemp()
+        workbook.save(file_path)
+        return file_path
+
+    @staticmethod
+    def export_records_for_user(data):
+        '''Export records for user.
+        Parameters
+        ------
+        data: list of dict{
+            event_name: string,
+            event_time: string,
+            event_location: string,
+            num_hours: string,
+            create_time: string,
+            role: string,
+            status: string,
+        }
+
+        Returns
+        ------
+        string
+            导出的excel文件路径
+        '''
+        if data is None:
```

Comment:
`data` could never be `None` unless you give a `None`, based on your function signature, `data` is a list, so there is no need to check `data` is `None`. Instead, you can check:
```
if not data:
    ...
```
it means you don't allow export for empty data.